pkg/security/probe/constantfetch/btfhub/constants.json

```diff
@@ -0,0 +1,9203 @@
+			"bpf_prog_attach_type_offset": 8,
+			"bpf_prog_aux_id_offset": 28,
+			"bpf_prog_aux_name_offset": 504,
+			"bpf_prog_aux_offset": 32,
+			"bpf_prog_tag_offset": 20,
+			"bpf_prog_type_offset": 4,
+			"creds_uid_offset": 4,
+			"dentry_sb_offset": 104,
+			"pid_level_offset": 4,
+			"pid_numbers_offset": 96,
+			"pipe_inode_info_bufs_offset": 152,
+			"sb_magic_offset": 96,
+			"sizeof_inode": 600,
+			"sizeof_upid": 16,
+			"tty_name_offset": 360,
+			"tty_offset": 400
+		},
+		{
+			"bpf_map_id_offset": 48,
+			"bpf_map_name_offset": 80,
+			"bpf_map_type_offset": 24,
+			"bpf_prog_attach_type_offset": 8,
+			"bpf_prog_aux_id_offset": 28,
+			"bpf_prog_aux_name_offset": 504,
+			"bpf_prog_aux_offset": 32,
+			"bpf_prog_tag_offset": 20,
+			"bpf_prog_type_offset": 4,
+			"creds_uid_offset": 4,
+			"dentry_sb_offset": 104,
+			"pid_level_offset": 4,
+			"pid_numbers_offset": 96,
+			"pipe_inode_info_bufs_offset": 152,
+			"sb_magic_offset": 96,
+			"sizeof_inode": 600,
+			"sizeof_upid": 16,
+			"tty_name_offset": 360,
+			"tty_offset": 408
+		},
+		{
+			"bpf_map_id_offset": 48,
+			"bpf_map_name_offset": 88,
+			"bpf_map_type_offset": 24,
+			"bpf_prog_attach_type_offset": 8,
+			"bpf_prog_aux_id_offset": 28,
+			"bpf_prog_aux_name_offset": 416,
+			"bpf_prog_aux_offset": 32,
+			"bpf_prog_tag_offset": 20,
+			"bpf_prog_type_offset": 4,
+			"creds_uid_offset": 4,
+			"dentry_sb_offset": 104,
+			"pid_level_offset": 4,
+			"pid_numbers_offset": 96,
+			"pipe_inode_info_bufs_offset": 152,
+			"sb_magic_offset": 96,
+			"sizeof_inode": 600,
+			"sizeof_upid": 16,
+			"tty_name_offset": 368,
+			"tty_offset": 400
+		},
+		{
+			"bpf_map_id_offset": 48,
+			"bpf_map_name_offset": 88,
+			"bpf_map_type_offset": 24,
+			"bpf_prog_attach_type_offset": 8,
+			"bpf_prog_aux_id_offset": 28,
+			"bpf_prog_aux_name_offset": 416,
+			"bpf_prog_aux_offset": 32,
+			"bpf_prog_tag_offset": 20,
+			"bpf_prog_type_offset": 4,
+			"creds_uid_offset": 4,
+			"dentry_sb_offset": 104,
+			"pid_level_offset": 4,
+			"pid_numbers_offset": 96,
+			"pipe_inode_info_bufs_offset": 152,
+			"sb_magic_offset": 96,
+			"sizeof_inode": 600,
+			"sizeof_upid": 16,
+			"tty_name_offset": 360,
+			"tty_offset": 400
+		},
+		{
+			"bpf_map_id_offset": 48,
+			"bpf_map_name_offset": 88,
+			"bpf_map_type_offset": 24,
+			"bpf_prog_attach_type_offset": 8,
+			"bpf_prog_aux_id_offset": 28,
+			"bpf_prog_aux_name_offset": 416,
+			"bpf_prog_aux_offset": 32,
+			"bpf_prog_tag_offset": 20,
+			"bpf_prog_type_offset": 4,
+			"creds_uid_offset": 4,
+			"dentry_sb_offset": 104,
+			"pid_level_offset": 4,
+			"pid_numbers_offset": 96,
+			"pipe_inode_info_bufs_offset": 152,
+			"sb_magic_offset": 96,
+			"sizeof_inode": 600,
+			"sizeof_upid": 16,
+			"tty_name_offset": 360,
+			"tty_offset": 408
+		}
+	],
+	"kernels": [
+		{
+			"distrib": "amzn",
+			"version": "2",
+			"arch": "arm64",
+			"uname_release": "4.14.101-91.76.amzn2.aarch64",
+			"cindex": 0
```

Comment:
Why are the constants in a separate array and not nested here?

Reply:
Because a lot of kernels have the same constants (`5.4.0.x-gcp` and `-aws` for example). This allows to have >1000 kernels in the array but only ~40 constant sets

Reply:
Gotcha, that makes sense.